src/System.Drawing.Common/src/misc/ClientUtils.cs

```diff
@@ -219,66 +41,38 @@ private static void Debug_NonSequentialEnumIsDefinedCheck(Enum value, int minVal
         /// Users of this class should set the RefCheckThreshold property 
         /// explicitly or call ScavengeReferences every once in a while to 
         /// remove dead references.
-        /// Also avoid calling Remove(item).  Instead call RemoveByHashCode(item)
+        /// Also avoid calling Remove(item). Instead call RemoveByHashCode(item)
         /// to make sure dead refs are removed.
-        /// -----------------------------------------------------------------
-        ///
         /// </summary>        
         internal class WeakRefCollection : IList
         {
-            private int _refCheckThreshold = Int32.MaxValue; // this means this is disabled by default.
-            private ArrayList _innerList;
+            internal WeakRefCollection() : this(4) { }
 
-            internal WeakRefCollection()
-            {
-                _innerList = new ArrayList(4);
-            }
+            internal WeakRefCollection(int size) => InnerList = new ArrayList(size);
 
-            internal WeakRefCollection(int size)
-            {
-                _innerList = new ArrayList(size);
-            }
-
-            internal ArrayList InnerList
-            {
-                get { return _innerList; }
-            }
+            internal ArrayList InnerList { get; }
 
             /// <summary>
             /// Indicates the value where the collection should check its items to remove dead weakref left over.
             /// Note: When GC collects weak refs from this collection the WeakRefObject identity changes since its 
-            ///       Target becomes null.  This makes the item unrecognizable by the collection and cannot be
+            ///       Target becomes null. This makes the item unrecognizable by the collection and cannot be
             ///       removed - Remove(item) and Contains(item) will not find it anymore.
+            /// A value of int.MaxValue means disabled by default.
             /// </summary>
-            public int RefCheckThreshold
-            {
-                get
-                {
-                    return _refCheckThreshold;
-                }
-                set
-                {
-                    _refCheckThreshold = value;
-                }
-            }
+            public int RefCheckThreshold { get; set; } = int.MaxValue;
 
             public object this[int index]
             {
                 get
                 {
-                    WeakRefObject weakRef = InnerList[index] as WeakRefObject;
-
-                    if ((weakRef != null) && (weakRef.IsAlive))
+                    if (InnerList[index] is WeakRefObject weakRef && weakRef.IsAlive)
```

Comment:
Your change is fine in that it keeps the same structure that was there before, so it's good to leave it as you have it for now.  But the weakRef.IsAlive check looks unnecessary, and the getter could just be:
```C#
get => (InnerList[index] as WeakRefObject)?.Target;
```
If the wrapped object has been collected, then Target will return null, which is the same thing this will return if IsAlive is false, and there's a race condition that has us hit that path anyway, since the object could be collected after the IsAlive check and before we access Target.